src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -384,6 +398,75 @@ type Currency {
   digits: Int!
 }
 
+"Trip on a specific date"
+type DatedTrip implements Node {
+  """
+  The service date when the trip occurs.
+    
+  **Note**: A service date is a technical term useful for transit planning purposes and might not
+    correspond to a how a passenger thinks of a calendar date. For example, a night bus running
+    on Sunday morning at 1am to 3am, might have the previous Saturday's service date.
+  """
+  date: LocalDate!
+  "The time when the trip ends including real-time information, if available."
+  end: ArrivalDepartureTime
+  "Global object ID provided by Relay. This value can be used to refetch this object using **node** query."
+  id: ID!
+  "The pattern the trip is running on"
+  pattern: Pattern
+  "The route the trip is running on"
+  route: Route!
+  "The time when the trip starts including real-time information, if available."
+  start: ArrivalDepartureTime
```

Comment:
Does this and the end also need to be an union to account for flex?

Reply:
Also, still unsure how we currently expose frequency based trips or how we should expose them.